scripts/cubs_tool.py

```diff
@@ -0,0 +1,421 @@
+#
+# MIT License
+#
+# (C) Copyright 2022-2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+import subprocess
+import json
+import rbd
+import rados
+import sys
+from argparse import ArgumentParser
+from prettytable import PrettyTable
+from packaging import version
+import time
+import os
+
+"""
+Start report block.
+Basic report of the cluster for upgrade purposes.
+Requires minimum podman version 3.4.4
+"""
+
+def print_function(print_data, service):
+  print(print_data)
+
+
+def fetch_status(service, cmd):
+  try:
+    cmd_results = cluster.mon_command(json.dumps(cmd), b'', timeout=5)
+  except:
+    print('something went wrong')
+
+  refined_results = json.loads(cmd_results[1])
+  table = PrettyTable(['Host', 'Daemon Type', 'ID', 'Version', 'Status', 'Image Name'])
+  table.title = service
+  for s in range(len(refined_results)):
+      host = (refined_results[s]["hostname"])
+      type = (refined_results[s]["daemon_type"])
+      id = (refined_results[s]["daemon_id"])
+      status = (refined_results[s]["status_desc"])
+      image_name = (refined_results[s]["container_image_name"])
+      if status == "running":
+        vers = (refined_results[s]["version"])
+      else:
+        vers = "unknown"
+      table.add_row([host, type, id, vers, status, image_name])
+  print_function(table, service)
+
+"""
+End report block.
+"""
+
+"""
+Start upgrade check and execute block.
+All functions related to checking for valid upgrade options or initiating an
+upgrade should live below.
+"""
+
+def fetch_base_current_vers():
+    cmd = {"prefix":"version", "format":"json"}
+    cmd_results = cluster.mon_command(json.dumps(cmd), b'', timeout=5)
+    results = json.loads(cmd_results[1])
+    for key,value in results.items():
+      current_version = str(value.split(' ')[2])
+    return current_version
+
+def fetch_curr_sha(cmd, new_version):
+    #cmd = {"prefix":"orch ps", "format":"json"}
```

Comment:
Can you remove this comment? it's a bit confusing.

Reply:
removed